src/components/forms/ProjectGrantsForm.tsx

```diff
@@ -415,132 +669,273 @@ export const ProjectGrantsForm: FC = () => {
-            closeMenuOnSelect={true}
-            selectedOptionColor='brand.option'
-            chakraStyles={chakraStyles}
-            onChange={handleReferralSource}
+          <Input
+            id='referrals'
+            type='text'
+            bg='white'
+            borderRadius={0}
+            borderColor='brand.border'
+            h='56px'
+            _placeholder={{ fontSize: 'input' }}
+            color='brand.paragraph'
+            fontSize='input'
+            mt={3}
+            {...register('referrals', {
+              maxLength: 150
+            })}
           />
-        </FormControl>
 
-        <Fade in={(referralSource as ReferralSource).value === OTHER} delay={0.25}>
-          <FormControl
-            id='referred-control'
-            mb={12}
-            display={(referralSource as ReferralSource).value === OTHER ? 'block' : 'none'}
-          >
-            <FormLabel htmlFor='referred' mb={1}>
-              <PageText fontSize='input'>
-                Did anyone recommend that you contact Ecosystem Support?
+          {errors?.referralSourceIfOther?.type === 'maxLength' && (
+            <Box mt={1}>
+              <PageText as='small' fontSize='helpText' color='red.500'>
+                Referral name cannot exceed 150 characters.
               </PageText>
-            </FormLabel>
-
-            <PageText as='small' fontSize='helpText' color='brand.helpText'>
-              Please write the name of the person who recommended that you apply.
-            </PageText>
-
-            <Input
-              id='referred'
-              type='text'
-              bg='white'
-              borderRadius={0}
-              borderColor='brand.border'
-              h='56px'
-              _placeholder={{ fontSize: 'input' }}
-              color='brand.paragraph'
-              fontSize='input'
-              mt={3}
-            />
-          </FormControl>
-        </Fade>
-
-        <Flex
-          bgColor='brand.uploadProposal'
-          justifyContent='space-evenly'
-          alignItems='center'
-          cursor='pointer'
-          py={9}
-          px={{ base: 6, md: 16 }}
-          mt={12}
-          mb={12}
-        >
-          <Box mr={6} flexShrink={0}>
-            <Image src={uploadSVG} alt='Upload file' height={42} width={44} />
-          </Box>
-
-          <Stack>
-            <PageText fontSize='input' fontWeight={400} lineHeight='21px' mb={-1}>
-              <strong>Upload the proposal.</strong> Click here or drag file to this box.
-            </PageText>
+            </Box>
+          )}
+        </FormControl>
 
-            <PageText as='small' fontSize='helpText' color='brand.helpText' lineHeight='17px'>
-              If you already have a proposal or document you&apos;d ike to share, please upload it
-              here. This is optional, but highly recommended.
-            </PageText>
-          </Stack>
-        </Flex>
+        <Controller
+          name='uploadProposal'
```

Comment:
Another issue I see is that:
- when you drop a file, it works as expected
- when you click on the box and select a file, it doesn't work